### DISCUSSION TITLE
Why derives from another class and this is not yet supported?

### DISCUSSION THREAD

**Discussion by CodingOctocat** [realm/realm-dotnet#3622] [Q&A] [answered]
Why derives from another class and this is not yet supported? Is it a compromise for current technical reasons or is it a design imperative? One of my model classes inherits `ObservableValidator`, which prevents me from using `IRealmObject`, I could use FluentValidation, but I prefer `System.ComponentModel.DataAnnotations`.

**Comment by nirinchev** [comment, marked as answer]
The problem is that Realm doesn't support inheritance, so the current design is a compromise that explicitly expresses this restriction. The problem is that if we allowed inheritance, it would imply that one could define model properties in a base class and have them show up in all inheritors. While this is somewhat difficult to achieve in practice, it's not impossible, but then it opens the door for using the base class as the type of property/list:

```csharp
partial abstract class PersonBase : IRealmModel
{
  [PrimaryKey]
  public ObjectId Id { get; set; }

  public string Name { get; set; }
}

partial class Teacher : PersonBase
{
}

partial class Student : PersonBase
{
}

partial class School : IRealmObject
{
  public IList<PersonBase> Members { get; }
}
```

The `Members` property would imply that both students and teachers can be added to the list, when in reality, that's not the case, due to the database not supporting polymorphism. While we do want to eventually make it work, it's not on our immediate todo list.
